src/highlight.js

```diff
@@ -787,6 +798,9 @@ const HLJS = function(hljs) {
     @returns {Language | never}
   */
   function requireLanguage(name) {
+    console.warn("requireLanguage is deprecated and will be removed entirely in the future.");
+    console.warn("Please see https://github.com/highlightjs/highlight.js/pull/2844");
```

Comment:
Should there be some sort of check here for checking whether highlight.js is running a production environment?

Reply:
The idea is that everyone can see it. It only shows up in console so it’s not disruptive. And after this PR it’s also immediately actionable so it’s not something that people have to wait till version 11. Someone can see it immediately make changes to that code base to make it go away. 

I’m pretty sure this is consistent with the other deprecation notices.

Reply:
I do certainly wish there were some better way to handle this but like I can’t really think of what it would be too of my head.

Reply:
I don't really feel strongly about this either way. My initial thought was someone using a plug-in in their CMS that had a bundled version of highlight.js or something similar and fixing the warning was beyond their control until the plug-in maintainer updated it.

Closest thing I can think of is webpack's support for `process.env.node_env === 'production'` but I can't say I've ever bothered looking for a browser-only solution other than having `*.production.min.js` and `*.development.js` files

Reply:
There is minified or not (perhaps could have a flag based on the build type) but there is no guarantee one is using minified in production and un-minified for development work.  Unless you're hacking on the library you're probably just using the minified version everywhere.